File: rec-sepval/src/test/java/net/kimleo/rec/sepval/parser/SimpleParserTest.java
package net.kimleo.rec.sepval.parser;

import net.kimleo.rec.sepval.SepValEntry;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class SimpleParserTest {


    private final SimpleParser parser = new SimpleParser();

    @Test
    public void shouldParseSimpleCSV() throws Exception {
        SepValEntry tuple = parser.parse("a, b, c");

        assertEquals(tuple.getSize(), 3);
        assertEquals(tuple.get(1), "b");

    }

    @Test
    public void shouldParseQuotedString() throws Exception {
        SepValEntry tuple = parser.parse("\"abc\",    \" \"\" \t\r\n\\\b def \"    , g");

        assertEquals(tuple.getSize(), 3);
        assertEquals(tuple.get(1), " \" \t\r\n\\\b def ");
        assertEquals(tuple.get(2), "g");

    }

    @Test
    public void shouldParseDifferentDelimiter() throws Exception {
        SepValEntry tuple = new SimpleParser(new ParseConfig('|'))
                .parse("hello|world|    123456|\"\"\"I have a dream\"\"\"");

        assertEquals(tuple.getSize(), 4);
        assertEquals(tuple.get(0), "hello");
        assertEquals(tuple.get(2), "123456");
        assertEquals(tuple.get(3), "\"I have a dream\"");
    }

    @Test
    public void shouldParseRecordFormat() throws Exception {
        SepValEntry tuple = parser.parse("first name, last name, {2}, ID, ..., comment");

        assertNotNull(tuple);

    }
}

File: src/test/kotlin/net/kimleo/rec/SimpleParserTest.kt
package net.kimleo.rec

import org.junit.Assert.*
import org.junit.Test

class SimpleParserTest {
    val parser = SimpleParser()

    @Test
    fun should_parse_single_csv() {
        val tuple = parser.parse("a, b, c")!!

        assertEquals(tuple.fields.size, 3)
        assertEquals(tuple.fields[1].value, "b")
    }

    @Test
    fun should_parse_quoted_string() {
        val tuple = parser.parse("\"abc\",    \" \"\" \t\r\n\\\b def \"    , g")!!

        assertEquals(tuple.fields.size, 3)
        assertEquals(tuple.fields[1].value, " \" \t\r\n\\\b def ")
        assertEquals(tuple.fields[2].value, "g")
    }

    @Test
    fun should_parse_different_delimiter() {
        val tuple = SimpleParser(Configuration('|')).parse("hello|world|    123456|\"\"\"I have a dream\"\"\"")!!

        assertEquals(tuple.fields.size, 4)
        assertEquals(tuple.fields[0].value, "hello")
        assertEquals(tuple.fields[2].value, "123456")
        assertEquals(tuple.fields[3].value, "\"I have a dream\"")
    }
}